src/iuse_actor.cpp

```diff
@@ -2208,9 +2208,16 @@ long ammobelt_actor::use( player &p, item &, bool, const tripoint& ) const
 void repair_item_actor::load( JsonObject &obj )
 {
     // Mandatory:
-    JsonArray jarr = obj.get_array( "materials" );
-    while( jarr.has_more() ) {
-        materials.emplace( jarr.next_string() );
+    JsonArray jarr_materials = obj.get_array( "materials" );
+    while(jarr_materials.has_more() ) {
+        materials.emplace(jarr_materials.next_string() );
+    }
+
+    JsonArray jarr_requirements = obj.get_array("requirements");
+    if (!jarr_requirements.empty()) {
```

Comment:
This check is redundant. The following loop will work just fine with an empty array.